src/render/mod.rs

```diff
@@ -131,6 +147,21 @@ impl<'world, 'state, 'a, 'b, 'c> BevyLinesRenderBackend<'world, 'state, 'a, 'b,
 
         color.map(|co| co.as_hsla_f32()).unwrap_or(default)
     }
+
+    fn drawing_enabled(&self, object: DebugRenderObject) -> bool {
+        match object {
+            DebugRenderObject::Collider(h, ..) => self
+                .context
+                .colliders
+                .get(h)
+                .map(|co| {
+                    let entity = Entity::from_bits(co.user_data as u64);
+                    self.global || self.visible.contains(entity)
```

Comment:
`self.global` takes precedence above the component, it would seem; Consider early-return in the function with a simple `if self.global { return true; }`, avoiding the rest of this lookup behaviour.